manual/parts/vspace.tex

```diff
@@ -109,22 +109,19 @@ \subsubsection{AArch32}
 \subsubsection{AArch64}
 
 Depending on configuration, Arm AArch64 processors have page-table structures with 3 or 4 levels.
-The VSpace object is therefore implemented either by the \obj{PageGlobalDirectory} object (4 level
-configs) or the \obj{PageUpperDirectory} object (3 level configs). To help with writing code that is
-generic in this distinction, \texttt{libsel4} provides the macro \texttt{seL4\_ARM\_VSpaceObject}
-that expands to the correct corresponding object. All intermediate paging structures are indexed by
-9 bits of the virtual address. Depending on configuration, the top-level object is indexed by either
-9 or 10 bits. The macro \texttt{seL4\_VSpaceIndexBits} makes this value available under a generic
-name. The table below shows the four-level configuration. On three-level configurations, the
-invocations for \texttt{seL4\_ARM\_VSpaceObject} capabilities are instead available on
-\obj{PageUpperDirectory} caps.
+Regardless of the configuration, the VSpace object is implemented by the \obj{PageTable} object.
+To help with writing code that is generic, \texttt{libsel4} provides the macro \texttt{seL4\_ARM\_VSpaceObject}
+that expands to the correct corresponding object across architectures. All intermediate paging
+structures are indexed by 9 bits of the virtual address. Depending on configuration, the
+top-level object is indexed by either 9 or 10 bits. The macro \texttt{seL4\_VSpaceIndexBits}
+makes this value available under a generic name. The table below shows the four-level configuration.
```

Comment:
Should we have another table for the three-level configuration for completeness?

Reply:
Isn't the three-level exactly the same, but with one level less? AFAIK the only difference used to be at which level the page upper directory pointed at, but now it's all unified it's just all the same type of object.

Code can be very level agnostic now, it can just create and map PageTable objcts on demand when a mapping fails.

Reply:
I just thought given there are only two possible configurations, having each one outlined in the table would not take up too much space will also being complete by showing every possible paging setup.

But it's just a minor point.